### ISSUE TITLE
Several discrepancies between the paper and the source code

### ISSUE THREAD

**Issue by jsadu826** [Luffy03/Large-Scale-Medical#30]
Dear author, I found several discrepancies between the paper and the source code.

1. The figures in the paper use 4 x 4 = 16 base crops, while in the code it's 3 x 3 = 9 base crops, which implementation should I follow to reproduce your work.

2. The paper says "we fed the features before and after augmentation into p_s and p_t, respectively", while the code feeds the feature after dropout into the student projector.

![Image](https://github.com/user-attachments/assets/abd32ef7-b0b0-4a1f-ac15-744efff817ae)

3. In the "Inter-volume contrast" part, the paper says "V_A and V_B are sampled from the same anatomical region, e.g., both from the abdomen
region or both from the chest region", and the code set V_A and V_B to two consecutive samples in a batch. But since the samples in the batch are randomly sampled from the whole dataset, how to ensure they come from the same anatomical region?

![Image](https://github.com/user-attachments/assets/dd286fbf-60b4-456b-b61a-136aa6dfd861)

4. When calculating inter and intra-volume losses, I understand that the cosine similarity scores should be mapped to [0, 1] to represent the overlapping ratios, but why use relu function rather than reversible operation like linear or sigmoid mapping?

![Image](https://github.com/user-attachments/assets/f8adc094-b006-4e34-8be5-6479134cab1c)

5. When calculating regularization loss, why not use the raw cosine similarity score, where -1 means complete discrepancy and 1 means complete alignment? Is it designed to align with the inter and intra-volume losses calculation? Also, why detach one of the features.

![Image](https://github.com/user-attachments/assets/0b075249-0951-4236-a261-a5afbeff642b)

Besides, if I want to train on about 40K volumes randomly sampled from the whole PreCT-160K dataset, 4 x 4 = 16 base crops of size 64 x 64 x 64, and do self-supervised training on 8 80GB H800 GPUs, what batch size, total num steps, lr, and lr warmup steps do you suggest?

Looking forward to your reply!

**Comment by Luffy03** [comment]
Thanks for your attention to our work, and so sorry for overlooking this issue.

1. In our conference version, we find 4*4 is slightly better than 3*3 while increasing the computation costs. for large-scale data, we revise it as 3*3. if you have enough sources, you can try 4*4
2. Drop out is a kind of feature augmentation
3. Please check the dataloader part, if you did not use dataset shuffle, data from the same region will be in organized order
4. Linear and sigmoid will change the value range, and relu only discards the negative part
5. (1) You can choose this way, but it would not be a significant difference. (2) Detach is a common tech in teacher-student modules and regularization.
